src/conventional/config.rs

```diff
@@ -204,11 +204,15 @@ pub(crate) fn host_info(git: &GitHelper) -> Result<HostOwnerRepo, Error> {
         let mut repository = None;
         if let Some(mut segments) = url.path_segments() {
             owner = segments.next().map(|s| s.to_string());
-            repository = segments
-                .next()
-                .map(|s: &str| s.trim_end_matches(".git").to_string());
         }
 
+        let remove = format!("/{:?}", owner);
+        repository = Some(url.path())
+                    .map(|s| s
+                            .trim_start_matches(&remove)
+                            .trim_end_matches(".git")
+                            .to_string());
```

Comment:
Thank you for this pull request.

```suggestion
        repository = Some(url.path()
            .trim_start_matches(&remove)
            .trim_end_matches(".git")
            .to_string());
```

I think it is better to wrap the expression in a `Some` then only the first part.

M

Reply:
@hdevalke 

Cool!!!, try to refactor it because this is my first time touching Rust